### PR TITLE
TProfile2Poly

### PR DESCRIPTION
**Pull request by imKuehlschrank**
Extending TH2Poly such as to make it behave like TProfile2D with the addition of being able to deal with polygonal bins.

             TH2
                |
                |
                +-------+------+------+-----+-----+-------------+
                        |      |      |     |     |             |
                       TH2C   TH2S   TH2I  TH2F  TH2D         TH2Poly
                                                  |             |
                                                  |             |
                                             TProfile2D    TProfile2Poly

The main goal of TProfile2Poly: Showing averages / standard deviation of measured values per polygonal bins.

**[EDIT]**
Proposed tutorial:
https://gist.github.com/imKuehlschrank/1efbbd4426d9bdfa9ceb27e09ae7e958
Similar style as https://root.cern/doc/master/th2polyEurope_8C.html

### CODE REVIEW

**Comment by dpiparo** [comment]
@phsft-bot build please.

**Comment by dpiparo** [comment]
Hi @imKuehlschrank , usually to showcase this kind of features we provide at least a Tutorial (see for example https://root.cern/doc/master/th2polyEurope_8C.html). In this context, a nice Tracker endcap of would look really nice. Would you be willing to write a tutorial if everything goes well with your PR?

**Comment by imKuehlschrank** [comment]
@dpiparo No problem, I'll create such a tutorial if everything goes well.

**Comment by imKuehlschrank** [comment]
@dpiparo Something failed. Locally after pulling form upstream I'm getting some problems with
`make[2]: *** No rule to make target '/home/fil/projects/root/root/math/genvector/inc/Math/LinkDef_Point3D.h', needed by 'math/genvector/G__GenVector.cxx'.  Stop.`
I'm not really sure if that's the problem with the ci, nor what I could've done to cause this locally. Any hints?

**Comment by dpiparo** [comment]
Hi @imKuehlschrank, can you try to reconfigure and rebuild in a different directory? If you have ccache you can enable it with the option -Dccache=ON to speed up re-compilation.

**Comment by imKuehlschrank** [comment]
@dpiparo your suggestion worked like a charm, thanks.
So what is the remaining problem with travis-ci, and is there anything I can do to fix it?

**Comment by dpiparo** [comment]
That are our coding conventions which can be enforced with clang-format and our configuration file which is located in the root top directory (.clang-format). Before proceeding with anything such as formatting, more doxygen documentation and the tutorial macro I would wait for @lmoneta and @couet to give you feedback. This will probably happen next week because of the Easter break.

**Comment by lmoneta** [comment]
@phsft-bot build please

**Comment by lmoneta** [comment]
Hi Filip, 

here are some comments. I am sorry I have no time to meet you today. 

 I see you are missing the implementation of TProfile2Poly:":GetBinError. 
This is needed otherwise the wrong error from TH2Poly is used

Also I would add TProfile2Poly::GetBinEntries using the similar function definer for the bins.

In the TProfile2Poly::Merge I see that the merge from overflows merge only the sum of the weights. I would addd a Merge (or Add) function in the TProfile2PolyBin class so you can re-use it for all bins of 
TProfile2Poly. 

 Thank you for updating your PR

 Best 

 Lorenzo

**Comment by lmoneta** [comment]
Hi Filip, 

 Thank you for the update. 
I see there is an issue with the computation of the bin error of TProfile2Poly in the default case (kMEANERROR). It looks like you are computing the inverse of the error.

I would strongly recommend to add in the test also a check on the computed bin errors, and maybe also bin entries and also the global statistical quantities (GetMean, GetStdDev, etc..) 

For example you could reuse the following functions used in the stressHistogram test, 

See https://github.com/root-project/root/blob/master/test/stressHistogram.cxx#L10594

and https://github.com/root-project/root/blob/master/test/stressHistogram.cxx#L10470

Lorenzo

**Comment by vgvassilev** [comment]
If you prefer to add a test using the google test infrastructure please have a look how it was done in PR #486

**Comment by Axel-Naumann** [comment]
PR #486 is not how we do Google test these days. Check https://github.com/root-project/root/commit/a5ca72d36bb5ba77e015f497858d623ef3f4bdd7 and e.g. https://github.com/root-project/root/blob/master/tree/treeplayer/test/array.cxx instead.

File: hist/hist/test/test_tprofile2poly.cxx [hunk 1]
```diff
@@ -0,0 +1,201 @@
+#include "gtest/gtest.h"
```
Comment (vgvassilev) [inline hist/hist/test/test_tprofile2poly.cxx]: Could you order the includes based on their generality.
First go ROOT related includes, then gtest and then STL.

**Pull request merged**